packages/nextra/src/server/mdx-plugins/rehype-mathjax-react.ts

```diff
@@ -0,0 +1,88 @@
+import type { Element, Root } from 'hast'
+import { toText } from 'hast-util-to-text'
+import { SKIP, visitParents } from 'unist-util-visit-parents'
+import type { VFile } from 'vfile'
+
+/** @type {Readonly<Options>} */
+const emptyOptions = {}
+/** @type {ReadonlyArray<unknown>} */
+const emptyClasses: string[] = []
+
+export type Options = Record<string, unknown> | undefined
+
+/**
+ * Render elements with a `language-math` (or `math-display`, `math-inline`)
+ * class with KaTeX.
+ *
+ * @param {Readonly<Options> | null | undefined} [options]
+ *   Configuration (optional).
+ * @returns
+ *   Transform.
+ */
+export function rehypeReactMathjax(options: Options) {
+  const _settings = options || emptyOptions
+
+  /**
+   * Transform.
+   */
+  return function (tree: Root, _file: VFile) {
+    visitParents(tree, 'element', function (element, parents) {
+      const classes = Array.isArray(element.properties.className)
```

Comment:
We can use `Set` here for classes instead of array

Reply:
I can change it, but it's probably overkill? I am not sure when sets become faster, but if most elements have 2 or 3 classes, I think arrays are actually faster.